x/payment/spec/docs.md

```diff
@@ -0,0 +1,80 @@
+## Abstract
+
+The payment module is responsible for paying for arbitrary data that will be added to the Celestia blockchain. While the data being submitted can be arbitrary, the exact placement of that data is important for the transaction to be valid. This is why the payment module utilizes a malleated transaction scheme. Malleated transactions allow for users to create a single transaction, that can later be malleated by the block producer to create a variety of different valid transactions that are still signed over by the user. To accomplish this, users to create a single `MsgWirePayForMessage` transaction, which is composed of metadata and signatures for multiple variations of the transaction that will be included onchain. After the transaction is submitted to the network, the block producer selects the appropriate signature and creates a valid `MsgPayForMessage` transaction depending on the square size for that block. This new malleated `MsgPayForMessage` transaction is what ends up onchain. 
+
+## State
+The only state that is modified is the sender’s account balance, via the bank keeper’s `Burn` method.
```

Comment:
Shouldn't the account nonce also be changed?

Reply:
That's a great point, at first I didn't think it was necessary because other modules don't mention it, but I don't see any reason not to be more explicit. https://github.com/celestiaorg/celestia-app/pull/187/commits/7e844e276d08fd9a69c720a72fd6b7b2c0799a5f